### ISSUE TITLE
Load test the Court's migration environment.

### ISSUE THREAD

**Issue by JessicaMarine** [ustaxcourt/ef-cms#138]
As the court, so that we can be assured that the migration environment can support expected capacity, we need to load test the migration environment. 

## Pre-Conditions
- [x] Use real legacy PDFs
- [x] System is loaded with lifelike cases & users & large PDFs
- [x] penetration testing can't happen simultaneously (coordinate with NSSPlus to ensure we aren't running this while they're doing pen testing)
- [x] ensure monitoring & alerting issue is complete (#250) so we can capture if/when the system fails - sync w/ @adunkman when things are nearly ready 

## Acceptance Criteria
* Production environment load testing completed
* Actionable items documented

## Security Considerations
* Does this work make you nervous about privacy or security?
* Does this work make major changes to the system?
* Does this work implement new authentication or security controls?
* Does this work create new methods of authentication, modify existing security controls, or explicitly implement any security or privacy features?

## Notes

* Elasticsearch may not autoscale, so special attention should be paid to its performance during load testing. Adjustments to its clustering, autoscaling, and instance type may be needed. 
* Elasticsearch-specific items are also tracked on Flexion’s Production Prep backlog as [Elasticsearch Performance - High](https://trello.com/c/ZAIu3KrT/581-elasticsearch-performance-high).

**Comment by adunkman** [comment]
Possible tools:
- https://gatling.io/ (@julialeague has experience)
- https://jmeter.apache.org/ (@adunkman has very limited experience)

**Comment by vickimcfadden** [comment]
Can't load test production enviro while pen testing is happening. So, in the meantime, we can load test migration environment.

**Comment by julialeague** [comment]
We ran a load test yesterday against the `migration` environment which was synchronized with ~50 court users using the front end at the same time. The test ramped up concurrent users over time, from about 20 concurrent users up to about 3,000, and from <100 requests per sec to >500 requests per second. This load level was decided on based on an estimate of the highest load we'd expect to see, plus some padding:

> I’m expecting peak traffic on the USTC site to be no greater than 1,500 users at once - basing that off digital analytics program data on other government website traffic. I took 1/3 of the 7 day total from irs.gov and distributed that across 7 days, then added in some padding

We managed to uncover some front-end issues in the process, mainly around the "loading" wheel becoming detached from the back end process it's tracking, leading to users needing to refresh a page to show that their action was completed. 

Back end performance appeared to stay relatively stable, despite the increased load - Mike observed the spike in traffic within our Kibana dashboard and our Cloudwatch monitoring, and the system had some CPU-related upticks, but nothing that was unmanageable. We also didn't observe any unknown errors in the api logs.
